pkg/yttlibrary/assert.go

```diff
@@ -25,6 +27,30 @@ var (
 
 type assertModule struct{}
 
+func (b assertModule) Equals(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 2 {
+		return starlark.None, fmt.Errorf("expected two arguments")
+	}
+
+	expected := args.Index(0)
+	actual := args.Index(1)
+	if _, notOk := expected.(starlark.Callable); notOk {
+		return starlark.None, fmt.Errorf("lambdas are not allowed")
+	}
+
+	expectedType := expected.Type()
+	actualType := actual.Type()
+	if expectedType != actualType {
+		return starlark.None, fmt.Errorf("arguments are of different types. expected %s, but got %s", expectedType, actualType)
+	}
+
+	if !reflect.DeepEqual(expected, actual) {
```

Comment:
This was covered elsewhere, reiterating here: `reflect.DeepEqual()` works for scalars, maps, arrays well, but it does not work for [`yamlFragments`](https://carvel.dev/ytt/docs/latest/lang-ref-yaml-fragment/#docs), so in order to support that use case as well we can encode the value in yaml, and compare them as strings that way.